### DISCUSSION TITLE
Side Modal Animation

### DISCUSSION THREAD

**Discussion by mattcullenmeyer** [twilio-labs/paste#2839] [General] [answered]
I'm wondering how the `animated` prop should work with `useSideModalState`. The side modal works as expected when implemented per the examples, but the side modal fails to successfully close when I attempt to add animation (see example below).

`const dialog = useSideModalState({});` <- this works
`const dialog = useSideModalState({ animated: true });` <- the modal won't close

I've seen other community discussions about using react-spring, but wasn't sure if that's something we need implement ourselves.

Thanks!

**Comment by ghost** [comment]
Thanks for the discussion here @mattcullenmeyer I'll spend a minute today looking into this

**Comment by ghost** [comment]
I was able to repro the issue and created a bug triage ticket for the team here: https://issues.corp.twilio.com/browse/DSYS-3668

**Comment by SiTaggart** [comment, marked as answer]
Hi @mattcullenmeyer 

The Side modal now comes with an animation, so there is no need to implement your own. Check it out here https://paste.twilio.design/components/side-modal and be sure to update to the latest version.

Thanks
